Fixtures/DependencyResolution/External/Branch/Bar/Package.swift

```diff
@@ -0,0 +1,26 @@
+// swift-tools-version:999.0
+// The swift-tools-version declares the minimum version of Swift required to build this package.
+
+import PackageDescription
+
+let package = Package(
+    name: "Bar",
+    products: [
+        // Products define the executables and libraries a package produces, and make them visible to other packages.
+        .library(
+            name: "Bar",
+            targets: ["Bar"]),
+    ],
+    dependencies: [
+        // Dependencies declare other packages that this package depends on.
+        // .package(url: /* package url */, from: "1.0.0"),
+	.package(url: "../Foo", branch: "master")
```

Comment:
Should be using `main`.  Or is that baked into the test fixture?  (which we should eventually fix too)

Reply:
That's baked into the fixture, it's just that the version of git the test uses hasn't been updated to use main instead of master

Reply:
Hmm... I wonder if that means that this test might unintentionally depend on the version of `git` that's installed.  It might make sense to do what some of the other tests do (e.g. `testMirrors`) which would be to add something like:
```
Process.checkNonZeroExit(args: "git", "-C", path.pathString, "init")
Process.checkNonZeroExit(args: "git", "-C", path.pathString, "checkout", "-b", "mybranch")
```
and then have the dependency use that.  That would make it independent of what the default branch was named.

Reply:
I think we have two kinds of test fixtures:  those that have a .zip file containing a `.git` with prebaked repository contents (branch names, tags, etc), and plain directories, which the `fixture()` function sets up to have a newly initialized repository.  For the latter kind, the name of the initial repository depends on the version of Git installed.

Maybe the right thing to do is to modify the `initGitRepo()` to check whether there's a `main` in the newly created repository and if not to do a:
```
Process.checkNonZeroExit(args: "git", "-C", path.pathString, "branch", "-m", "main")
```
so that all our unit tests can start out in a known state regardless of the version of `Git` installed on the machine running the test.

Reply:
Then your new unit test wouldn't have to be modified (though some others might have to be adjusted to account for it).  But it seems suboptimal that these tests will start failing once versions of Git that default to main start getting deployed more widely.

Reply:
What if we add 
```
try systemQuietly([Git.tool, "-C", dir.pathString, "checkout", "-b", "main"])
```
to `initGitRepo()` ensuring there's always a main branch?

Reply:
That sounds like a good idea.  We could just do:
```
branch -m main
```
though.  I tested that that is a no-op if the branch is already named `main`.

Reply:
But in general I agree:  could you modify `initGitRepo()` to do the `branch -m main`?  That would make all the tests a lot more robust.  Thanks!

Reply:
Yep, just got it working with `branch -m main`!

Reply:
Great!  Thank you!